File: apps/backend/models/outreach_draft.py
"""Mirrors packages/server/src/models/OutreachDraft.model.ts — same
"outreachdrafts" collection. Every draft starts "pending" and must be
explicitly approved/edited/rejected before anything is ever sent."""

from __future__ import annotations

from datetime import UTC, datetime
from typing import Literal

from beanie import Document, PydanticObjectId
from pydantic import BaseModel, Field
from pymongo import IndexModel

DraftStatus = Literal["pending", "approved", "edited", "rejected"]
HookConfidence = Literal["high", "low"]


class OutreachDraftEditedText(BaseModel):
    coldEmail: str | None = None
    linkedinNote: str | None = None
    referralRequest: str | None = None
    coverLetter: str | None = None


class OutreachDraft(Document):
    userId: PydanticObjectId
    jobIntelId: PydanticObjectId
    contactId: PydanticObjectId
    contactName: str
    hook: str
    hookConfidence: HookConfidence
    coldEmail: str
    linkedinNote: str
    referralRequest: str | None = None
    coverLetter: str | None = None
    status: DraftStatus = "pending"
    editedText: OutreachDraftEditedText | None = None
    createdAt: datetime = Field(default_factory=lambda: datetime.now(UTC))
    updatedAt: datetime = Field(default_factory=lambda: datetime.now(UTC))

    class Settings:
        name = "outreachdrafts"
        indexes = [IndexModel([("jobIntelId", 1), ("createdAt", -1)])]

File: packages/shared/src/types/outreachDraft.ts
/**
 * AI-drafted outreach for a specific contact, grounded in a job analysis
 * (JobIntel) and — when available — the candidate's own resume. Every
 * draft starts "pending" and must be explicitly approved, edited, or
 * rejected by the user before anything is ever sent (Phase 4). Ported
 * from HireConnect's outreach_content_agent + human_approval_gate.
 */
export type DraftStatus = "pending" | "approved" | "edited" | "rejected";
export type HookConfidence = "high" | "low";
/** "referral" assumes an existing connection and asks for a referral instead
 * of the standard cold-email/connection-note pair. */
export type OutreachDraftIntent = "outreach" | "referral";

export interface OutreachDraftEditedText {
  coldEmail?: string;
  linkedinNote?: string;
  referralRequest?: string;
  coverLetter?: string;
}

export interface OutreachDraft {
  id: string;
  userId: string;
  jobIntelId: string;
  contactId: string;
  contactName: string;
  /** The real, verifiable fact the draft references — never an invented detail. */
  hook: string;
  hookConfidence: HookConfidence;
  coldEmail: string;
  /** Deterministically truncated to 300 characters after generation — the model isn't fully reliable at respecting a hard limit on its own. */
  linkedinNote: string;
  /** Null until warm-path contact detection exists (Phase 3 / lead_discovery). */
  referralRequest: string | null;
  /** Null when the user has no resume on file. */
  coverLetter: string | null;
  status: DraftStatus;
  editedText: OutreachDraftEditedText | null;
  createdAt: string;
  updatedAt: string;
}
